packages/salesforce-adapter/src/change_validators/flow_referenced_elements.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Salto Labs Ltd.
+ * Licensed under the Salto Terms of Use (the "License");
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+
+import {
+  ChangeValidator,
+  getChangeData,
+  isAdditionOrModificationChange,
+  isInstanceChange,
+  InstanceElement,
+  ChangeError,
+  ElemID,
+  FieldMap,
+} from '@salto-io/adapter-api'
+import { TransformFuncSync, transformValuesSync } from '@salto-io/adapter-utils'
+import _ from 'lodash'
+import { isInstanceOfTypeSync } from '../filters/utils'
+import { FLOW_METADATA_TYPE, FLOW_NODE, TARGET_REFERENCE } from '../constants'
+
+type ReferenceToElemId = {
+  referenceOrName: string
+  elemId: ElemID
+  kind?: string
```

Comment:
Can kind ever be a string other than FLOW_NODE or TARGET_REFERENCE?
If not, it might be clearer to make this explicit.
```suggestion
  kind?: typeof FLOW_NODE | typeof TARGET_REFERENCE
```